Michael ayele prefixing prrte binaries

Michael-uofl opened the pull request:
Updated Makefile.am files to include @PRTE_BINARY_PREFIX@ in library
and binary names

jsquyres committed:
prte_configure_options.m4: Add --with-prte-binary-prefix

Add configure CLI option to allow optionally prefixing PRTE binary and
library filenames.

Signed-off-by: Jeff Squyres <jeff@squyres.com>

jsquyres committed:
prted: support --with-prte-binary-prefix

Add the prefix to the prted binary name that was passed in via
--with-prte-binary-prefix.

Signed-off-by: Jeff Squyres <jeff@squyres.com>

Michael-uofl committed:
Add prefixing capability to prrte library and binaries

Updated Makefile.am files to include @PRTE_BINARY_PREFIX@ in library and
binary names where appropriate.
Verified successful build and functionality of prefixed binaries
This commit ensures compatiblity with prefixed binary naming while
maintaining the integrity of the build system and runtime.

Signed-off-by: Michael Ayele <whayel01@louisville.edu>

jsquyres commented on src/tools/prted/Makefile.am:
```diff
@@ -51,4 +51,4 @@ prted_LDADD = \
     $(prte_libevent_LIBS) \
```
Missed adding the prefix to the 2 lines above this -- I'll fix.

jsquyres commented on config/prte_configure_options.m4:
```diff
@@ -377,4 +378,21 @@ fi
+    [AS_HELP_STRING([--with-prte-binary-prefix=STRING],
+                    ["Optionally add a prefix to all PRTE binary names (executables and libraries) names"])])
+#Set default prefix to "ompi-"
+AS_IF([with_prte_binary_prefix=ompi-])
```
This statement is incorrect (i.e., it's incomplete).  I'll fix.

jsquyres commented on src/Makefile.am:
```diff
@@ -43,15 +43,16 @@ DIST_SUBDIRS = \
+lib@PRTE_BINARY_PREFIX@prrte_la_DEPENDENCIES = \
+	util/lib@PRTE_BINARY_PREFIX@prrteutil.la \
 	$(MCA_prte_FRAMEWORK_LIBS)
 libprrte_la_LDFLAGS = -version-info $(libprrte_so_version)
 libprrte_la_CPPFLAGS =
```
Missed updating these, too.  I'll fix.

jsquyres committed:
Restore prte command

Commit c299264e88 removed the prte executable.  Turns out that we
actually need that command, so bring it back.

Signed-off-by: Jeff Squyres <jeff@squyres.com>

jsquyres commented:
* Restored the `prte` command
* Fixed some places where `PRTE_BINARY_PREFIX` was missed
* Fixed a mistake in the `.m4` file
* Took the upstream fix from https://github.com/openpmix/prrte/pull/2087

@hppritcha Please review

hppritcha commented:
trying to build this branch and autogen.pl is failing
```
libtoolize: copying file 'config/ltsugar.m4'
libtoolize: copying file 'config/ltversion.m4'
libtoolize: copying file 'config/lt~obsolete.m4'
autoreconf: running: /usr/bin/autoconf --include=config --include=config/oac --force --warnings=all,no-obsolete,no-override
autoreconf: running: /usr/bin/autoheader --include=config --include=config/oac --force --warnings=all,no-obsolete,no-override
autoheader: cannot rename /tmp/aramcX1V/ahjFCanl/config.hin as 
autoheader: 
autoheader:     include/prte_version.h
autoheader: .in: No such file or directory
autoreconf: /usr/bin/autoheader failed with exit status: 1
Command failed: autoreconf -ivf --warnings=all,no-obsolete,no-override -I config -I config/oac
```

hppritcha commented:
pushing a small fix to this branch shortly.  Please squash it before merging.

hppritcha reviewed (approved):
I approve assuming the SQUASHME commit is included.

jsquyres commented:
@hppritcha There was a standalone commit that made that `configure.ac` change -- I just dropped it (and your SQUASHME).  Should be good now.
